### PR TITLE
separate scroll until visible implementation for lower level accessibility 

### PR DESCRIPTION
**Pull request by CareF**
## Description

This is a follow up of #62097, where in many use cases the `Scrollable` is not exposed: e.g. it's within `ListView`. so having separate API at lower level is useful. 

## Related Issues

#61458

## Tests

I modified `'scrollUntilVisible: scrolls to make unbuilt widget visible'` group in `packages/flutter_test/test/controller_test.dart` and added `'Drag Until Visible'` test.

## Checklist

Before you create this PR, confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter_test/lib/src/controller.dart [hunk 1]
```diff
@@ -817,26 +817,32 @@ abstract class WidgetController {
   /// Shorthand for `Scrollable.ensureVisible(element(finder))`
   Future<void> ensureVisible(Finder finder) => Scrollable.ensureVisible(element(finder));
 
-  /// Repeatedly scrolls the `scrollable` by `delta` in the
+  /// Repeatedly scrolls a [Scrollable] by `delta` in the
   /// [Scrpllable.axisDirection] until `finder` is visible.
```
Comment (liyuqian) [inline packages/flutter_test/lib/src/controller.dart:821]: Typo of the old code `Scrpllable` -> `Scrollable`.

**Review by liyuqian — COMMENTED** [review]
I'm not sure I fully understand "lower level accessibility" in this context. It seems that this PR is just helping us to save some extra `find.byType(Scrollable),` as illustrated in the unit test changes?

Is there a specific example where "the `Scrollable` is not exposed" and simply writing `find.byType(Scrollable),` in the old code wouldn't work?

**Comment by CareF** [comment]
> I'm not sure I fully understand "lower level accessibility" in this context. It seems that this PR is just helping us to save some extra `find.byType(Scrollable),` as illustrated in the unit test changes?
> 
> Is there a specific example where "the `Scrollable` is not exposed" and simply writing `find.byType(Scrollable),` in the old code wouldn't work?

The major reason I add this PR is for `dragUntilVisible`, which can be used even without finding the scrollable, at the price of needing an `Offset` rather than just a double as distance. @liyuqian

File: packages/flutter_test/lib/src/controller.dart (hunk 1 above)
Comment (CareF) [reply packages/flutter_test/lib/src/controller.dart]: Done.

**Comment by CareF** [comment]
@liyuqian the new test is ready.

File: packages/flutter_test/lib/src/controller.dart [hunk 2]
```diff
@@ -817,26 +817,32 @@ abstract class WidgetController {
...
   /// Throws a [StateError] if `finder` is not found for maximum `maxScrolls`
   /// times.
   ///
   /// This is different from [ensureVisible] in that this allows looking for
   /// `finder` that is not built yet, but the caller must specify the scrollable
-  /// that will build child specified by `finder`.
+  /// that will build child specified by `finder` when there are multiple
+  ///[Scrollable]s.
+  ///
+  /// See [dragUntilVisible].
```
Comment (liyuqian) [inline packages/flutter_test/lib/src/controller.dart:835]: Nit: `See also [dragUntilVisible].`

**Review by liyuqian — APPROVED** [review]
LGTM

File: packages/flutter_test/lib/src/controller.dart (hunk 2 above)
Comment (CareF) [reply packages/flutter_test/lib/src/controller.dart]: Done.

**Pull request merged**